clang/include/clang/Frontend/ASTUnit.h

```diff
@@ -116,7 +116,7 @@ class ASTUnit {
   std::shared_ptr<Preprocessor>           PP;
   IntrusiveRefCntPtr<ASTContext>          Ctx;
   std::shared_ptr<TargetOptions>          TargetOpts;
-  std::shared_ptr<HeaderSearchOptions>    HSOpts;
+  std::unique_ptr<HeaderSearchOptions> HSOpts;
```

Comment:
Is there still a reason to keep this a pointer?

Reply:
Not really, I just didn't want to change the size of `ASTUnit` drastically (`HeaderSearchOptions` is 576B).

Reply:
(And didn't want to pay the initialization cost unconditionally.)